datafusion/physical-expr/src/aggregate/array_agg_distinct.rs

```diff
@@ -153,12 +153,11 @@ impl Accumulator for DistinctArrayAggAccumulator {
             return Ok(());
         }
 
-        let array = &states[0];
-
-        assert_eq!(array.len(), 1, "state array should only include 1 row!");
```

Comment:
Is it okay that we just delete the assertion?

Reply:
yes, because it is no longer true, this PR is dealing with the array that has len more than one